includes/class-amp-story-post-type.php

```diff
@@ -105,6 +105,39 @@ public static function register() {
 			]
 		);
 
+		$meta_args = [
+			'type'           => 'array',
+			'object_subtype' => self::POST_TYPE_SLUG,
+			'single'         => true,
+			'show_in_rest'   => [
+				'schema' => [
+					'type'  => 'array',
+					'items' => [
+						'id'       => [
+							'type' => 'string',
+						],
+						'type'     => [
+							'type'    => 'string',
+							'default' => 'page',
+						],
+						'elements' => [
+							'type'    => 'array',
+							'default' => [],
+						],
+						'index'    => [
+							'type'    => 'integer',
+							'default' => 0,
+						],
+					],
+				],
+			],
+		];
+		// Hide the pages data from none logged in users.
+		if ( ! is_user_logged_in() ) {
+			$meta_args['show_in_rest'] = false;
+		}
+		register_meta( 'post', 'amp_pages', $meta_args );
```

Comment:
I wonder if we should register a more generic `post_meta` instead of `amp_pages` (e.g. `story_content`) and make `amp_pages` a part of it?

Reason: there might be additional information that might not necessarily be related to a specific page (general settings), such as `Advance to next Page` setting in the Legacy Stories. There might potentially be more of those in the future. Or do you think it would make more sense to register a separate meta for each new setting? Or register a separate meta for all generic settings? Thoughts?

Reply:
I agree with this. For future compatibility we shouldn't (through naming) restrict what to save in the field. `story_content` (or `amp_story_content`) is a candidate.

Reply:
I think we should still prefix our key with something, just in case someone else tries to use `story_content`. I can work with `amp_story_content`. 

This meta key is only for the pages JSON object. Other meta data like deep linking etc, would be stored in another object / meta key.

Reply:
Now we are using `post_content_filtered`, resolving.